toonz/sources/toonz/vectorinspector.cpp

```diff
@@ -0,0 +1,678 @@
+#include "toonz/tcolumnhandle.h"
+#include "toonz/txsheethandle.h"
+#include "toonz/txshcell.h"
+
+#include "tvectorimage.h"
+#include "toonz/tvectorimageutils.h"
+#include "tstroke.h"
+#include "toonz/tobjecthandle.h"
+#include "toonzqt/selection.h"
+#include "toonzqt/tselectionhandle.h"
+#include "tools/tool.h"
+#include "tools/toolhandle.h"
+#include "tools/strokeselection.h"
+#include "../tnztools/vectorselectiontool.h"
+
+#include <QApplication>
+#include <QPainter>
+#include <QScrollBar>
+#include <QFrame>
+#include <QPaintEvent>
+#include <QVBoxLayout>
+#include <QLabel>
+#include <QDateTime>
+#include <QDebug>
+
+#include <QMutex>
+
+#include <QtWidgets>
+
+//-----------------------------------------------------------------------------
+VectorInspectorFrame::VectorInspectorFrame(QScrollArea *parent,
+                                           Qt::WindowFlags flags)
+    : QFrame(parent, flags), m_scrollArea(parent) {
+  setObjectName("VectorInspectorFrame");
+  setFrameStyle(QFrame::StyledPanel);
+
+  setFocusPolicy(Qt::StrongFocus); /*-- Keyboard Tab Focus --*/
+
+  setFixedHeight(parentWidget()->height());
+  setMinimumWidth(std::max(parentWidget()->width(), 800));
+}
+
+//------------------------------------------------------------------------------
+
+VectorInspectorPanel::VectorInspectorPanel(QWidget *parent,
+                                           Qt::WindowFlags flags)
+    : QWidget(parent) {
+  initiatedByVectorInspector   = false;
+  initiatedBySelectTool        = false;
+  strokeOrderChangedInProgress = false;
+  selectingRowsForStroke       = false;
+
+  m_field = new QLabel("I am a test value for the QLabel");
+
+  m_field->setFixedHeight(16);
+
+  std::vector<int> m_selectedStrokeIndexes = {};
+
+  setFocusProxy(m_field);
+
+  proxyModel = new MultiColumnSortProxyModel;
+
+  proxyView = new QTreeView;
+  proxyView->setRootIsDecorated(false);
+  proxyView->setAlternatingRowColors(true);
+  proxyView->setModel(proxyModel);
+  proxyView->setSortingEnabled(true);
+
+  proxyView->setSelectionMode(QAbstractItemView::ExtendedSelection);
+  proxyView->setSelectionBehavior(QAbstractItemView::SelectRows);
+
+  proxyView->setContextMenuPolicy(Qt::CustomContextMenu);
+  connect(proxyView, &QWidget::customContextMenuRequested, this,
+          &VectorInspectorPanel::showContextMenu);
+
+  filterPatternLineEdit = new QLineEdit;
+  filterPatternLabel    = new QLabel(tr("&Filter pattern (example 0|1):"));
+  filterPatternLabel->setBuddy(filterPatternLineEdit);
+
+  filterSyntaxComboBox = new QComboBox;
+  filterSyntaxComboBox->addItem(tr("Regular expression"), QRegExp::RegExp);
+  filterSyntaxComboBox->addItem(tr("Wildcard"), QRegExp::Wildcard);
+  filterSyntaxComboBox->addItem(tr("Fixed string"), QRegExp::FixedString);
+  filterSyntaxLabel = new QLabel(tr("Filter &syntax:"));
+  filterSyntaxLabel->setBuddy(filterSyntaxComboBox);
+
+  filterColumnComboBox = new QComboBox;
+  filterColumnComboBox->addItem(tr("#"));
```

Comment:
I suggest changing this to "Stroke #" to make it clear to a user what this number represents.

Reply:
It was Stroke originally, but that kept overlapping the sort order triangle. I'll take another look at this. Qt 5.15.2 seems to have a more prominent triangle and I may learn the way to control spacing to include space for the triangle.

Reply:
The first column:
Stroke #, can be confused with the Id column which is the Stroke ID. The Stroke ID persists across movement up/down the list of strokes making it a way to track a stroke in discussion. 

It is an ordered Position value in the list of strokes from Back (0) to Front (highest value). A label of Position, or Pos. #. , Order #, or no label at all may be clearest. It is similar to the row number on a spreadsheet border in that the numbers are ascending or descending representing the stroke data in the position from front to back, or back to front.
Here is what I looks like with no label on that column:
![image](https://github.com/tahoma2d/tahoma2d/assets/44948041/0afeaeb3-e60e-4613-9f04-d9abb0a0c8f9)

Reply:
`Stroke Order` is probably most appropriate

I believe I know what `Quad` is supposed to be, though not sure those who haven't looked at the code will know what those are supposed to be. What is `P`?

Reply:
P0, P1, P3 are the symbols used in the Quadratic equation formula. 
![image](https://github.com/tahoma2d/tahoma2d/assets/44948041/baf7b074-4d87-4620-9320-ffc3442191e3)

FYI: I added a markdown formatted write-up describing VectorInspector to the first message of this Pull Request.

Reply:
With the heading Stroke Order and the sort indicator:
![image](https://github.com/tahoma2d/tahoma2d/assets/44948041/b67c2946-a962-4e5b-9f10-8551cb80ea27)

Reply:
This has been implemented.